website/blog/2023-03-30-relay-15.mdx

```diff
@@ -0,0 +1,153 @@
+  }
+}
+```
+
+And for users who are using file-based persisted queries a new feature flag, `compact_query_text` was added to remove all whitespace from persisted query output text. This can cause a substantial reduction in your persisted query text file (we've seen an example of reducing a single query text from 84kb to 31kb). This new feature flag can be enabled within your Relay config file.
+
+```js
+persistConfig: {
+  file: 'path/to/file.json',
+  algorithm: 'SHA256'
+},
+featureFlags: {
+  compact_query_text: true
+}
+```
+
+### Typesafe updates now support missing field handlers
+
+One of the few breaking changes in this release, the [missing field handlers](https://relay.dev/docs/guided-tour/reusing-cached-data/filling-in-missing-data) function signature has been changed. The `record` argument to the handler used to recieve a `Record` type (which is an untyped grabbag containing whatever the store has) but we now pass a `ReadOnlyRecordProxy` which is much more typesafe and provides a better interface. Furthermore, we also replaced the `field` argument of type `NormalizationLinkedField` with `CommonLinkedField`, which both a `ReaderLinkedField` and `NormalizationLinkedField` "implement".
+
+### Flow type improvements
+
+Flow users will now get types inferred from `graphql` literals with more Relay APIs. No longer do Flow users need to explicitly type the return value of the `usePreloadedQuery`, `useQueryLoader`, `useRefetchableFragment`, `usePaginationFragment`, and `useBlockingPaginationFragment` API methods.
+
+## What's new in the Relay VSCode Extension v2?
+
+The latest version of the [Relay VSCode Extension](https://marketplace.visualstudio.com/items?itemName=meta.relay) adds support for workspaces which contain multiple Relay configs in various subdirectories. Instead of only supporting a single Relay project at a time, you can now configure the `relay.projects` configuration to point to multiple Relay projects in a single workspace. This allows you to seamlessly switch between multiple projects and still be able to get field autocompletion, go to definition on fragments spreads, etc. And with the `relay.autoStartCompiler` you can have the Relay compiler run in watch mode on all projects in your workspace in parallel.
+
+If you were previously using the `relay.rootDirectory` or `relay.pathToConfig` configuration options, those have been deprecated and you'll want to migrate those to the `relay.projects` configuration instead.
+
+Before:
+
+```json
+// VSCode Settings
+"relay.rootDirectory": "path/to/app",
+"relay.pathToConfig": "config/relay.config.js"
+```
+
+After:
+
+```json
+// VSCode Settings
+"relay.projects": [{
+  "name": "my-app",
+  "rootDirectory": "path/to/app",
+  "pathToConfig": "config/relay.config.js"
+}]
+```
+
+Note that your existing settings will continue to work in this release, but you should migrate to the new settings as they'll be removed in a future release.
+
+## What's next?
+
+TODO..
```

Comment:
Not sure if there's any other roadmap based items we want to talk about here, if not we can easily remove.

Reply:
I think we can probably leave this off